From 6775cc94517696abe12bbae601395b7a16e06ba0 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Mon, 1 Aug 2016 10:52:46 +0200
Subject: [PATCH 55/55] enable cache=unsafe for vma extract_content and
 qmp_savevm_start

We don't send any flush here, so we need to open with cache=unsafe.

Signed-off-by: Alexandre Derumier <aderumier@odiso.com>
For-patch: BDRV_O_CACHE_WB was removed
---
 savevm-async.c | 2 +-
 vma.c          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/savevm-async.c b/savevm-async.c
index 2c41b17..6a2266c 100644
--- a/savevm-async.c
+++ b/savevm-async.c
@@ -253,7 +253,7 @@ void qmp_savevm_start(bool has_statefile, const char *statefile, Error **errp)
 {
     Error *local_err = NULL;
 
-    int bdrv_oflags = BDRV_O_RDWR;
+    int bdrv_oflags = BDRV_O_RDWR | BDRV_O_NO_FLUSH;
     int ret;
 
     if (snap_state.state != SAVE_STATE_DONE) {
diff --git a/vma.c b/vma.c
index 0716a45..4d2d5ad 100644
--- a/vma.c
+++ b/vma.c
@@ -280,7 +280,7 @@ static int extract_content(int argc, char **argv)
         } else if (di) {
             char *devfn = NULL;
             const char *format = NULL;
-            int flags = BDRV_O_RDWR;
+            int flags = BDRV_O_RDWR | BDRV_O_NO_FLUSH;
             bool write_zero = true;
 
             if (readmap) {
-- 
2.1.4

